### PR TITLE
integrative unit test

### PR DESCRIPTION
**Pull request by aazhou1**
## Summary by CodeRabbit

- **New Features**
	- Introduced a comprehensive testing framework for USDC token integration with the term repo strategy.
	- Added functionality to simulate auction processes and validate token management.

- **Bug Fixes**
	- Enhanced access control for auction offer submissions.

- **Tests**
	- Implemented multiple test methods to ensure proper functionality and performance of the integration.

### CODE REVIEW

File: src/test/TestUSDCIntegration.t.sol [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+        repoToken1WeekAuction.auctionSuccess(offerIds, fillAmounts, repoTokenAmounts);
+
+        holdings = termStrategy.repoTokenHoldings();
+
+        // test: 1 holding because auctionClosed not yet called
+        assertEq(holdings.length, 1);
+
+        termStrategy.auctionClosed();
+        holdings = termStrategy.repoTokenHoldings();
+        assertEq(holdings.length, 2);
```
Comment (0xddong) [inline src/test/TestUSDCIntegration.t.sol:114]: can we check here if the repo tokens are properly sorted by maturity?

File: src/test/TestUSDCIntegration.t.sol [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+        bytes32[] memory offers = termStrategy.pendingOffers();        
+
+        assertEq(offers.length, 0);
+
+        // assertEq(termStrategy.totalLiquidBalance(), initialState.totalLiquidBalance - 1e6);
```
Comment (0xddong) [inline src/test/TestUSDCIntegration.t.sol:119]: are these asserts failing?

File: src/test/TestUSDCIntegration.t.sol (hunk 1 above)
Comment (aazhou1) [reply src/test/TestUSDCIntegration.t.sol:114]: good idea

File: src/test/TestUSDCIntegration.t.sol (hunk 2 above)
Comment (aazhou1) [reply src/test/TestUSDCIntegration.t.sol:119]: will try it

**Pull request merged**